iroh-sync/src/sync.rs

```diff
@@ -838,6 +838,11 @@ impl RecordIdentifier {
         &self.0
     }
 
+    /// Get this [`RecordIdentifier`] as [Bytes].
+    pub fn to_bytes(&self) -> Bytes {
```

Comment:
should be `as_bytes` following the convention I believe

Reply:
`as_bytes` already exists and returns `&[u8]`. This `to_bytes` returns [`Bytes`] to skip the extra allocation when recreating a `Bytes`. Open for better ideas on naming, but `as_bytes` is taken..

Reply:
could do 
- `impl AsRef<[u8]>` for the `&[u8]` and
- `fn as_bytes(&self) -> Bytes`

Reply:
sounds good, changed to that